react/Coupon.tsx

```diff
@@ -17,63 +15,53 @@ const Coupon: StorefrontFunctionComponent<CouponProps> = () => {
     setCoupon(newCoupon)
   }
 
-  const closeCouponInput = () => {
+  const resetCouponInput = () => {
     setCoupon('')
     toggle()
```

Comment:
You should effectivelly set the value of the state. Toggling like this is bug prone.

Reply:
done!